keps/sig-api-machinery/4793-revise-apf-default-configuration/README.md

```diff
@@ -0,0 +1,971 @@
+-->
+
+Many features have been developed and use cases surfaced since the introduction
+of [KEP-1040]. These prompt us to revise APF default configuration to better
+make use of APF for better performance and reliability of API Server.
+
+For example, we have learned some priority levels are more critical than other
+priority levels as when critical requests fail, the whole Kubernetes control
+plane may become unavailable (e.g. node-controller become unavailable and no new
+Nodes can be registered due to node-controller's leader election requests being
+failed). For such priority levels, we propose to increase their nominal shares
+to avoid failing critical requests and lendable percentage as well to avoid
+wasting shares.
+
+We also have learned that Event requests are plentiful in upscaling but they are
+resilient to unavailability of API Server. So we can deprioritize them when API
+Server is under stress.
+
+[KEP-1040]: https://github.com/kubernetes/enhancements/blob/97713189b3107b41c4c19505d04aa7ef22df063b/keps/sig-api-machinery/1040-priority-and-fairness/README.md
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Better overload protection in perspective of whole Kubernetes control plane
+    instead of just API Server alone.
+-   More performant in cluster upscaling (e.g. increasing node count, deploying
+    pods)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+-   There will no fundamental changes in APF's own implementation
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This KEP proposes the following settings:
+
+Name            | Nominal Shares | Lendable | Proposed Borrowing Limit
+--------------- | -------------: | -------: | -----------------------:
+exempt          | 0              | 0%       | none
+leader-election | 50             | 80%      | none
+node-high       | 60             | 50%      | none
+system          | 40             | 50%      | none
+workload-high   | 40             | 50%      | none
+workload-low    | 40             | 75%      | none
+global-default  | 10             | 50%      | none
+catch-all       | 5              | 0%       | none
+
+In addition to above, a new priority level and flow schema will be created for
+Events:
+
+```
+apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
+kind: FlowSchema
+metadata:
+  name: events
+spec:
+  distinguisherMethod:
+    type: ByUser
+  matchingPrecedence: 450
+  rules:
+  - nonResourceRules:
+    resourceRules:
+    - apiGroups:
+      - '*'
```

Comment:
I think that the matching should be restricted to matching Kubernetes Event objects.